### PR TITLE
fix incorrect y-axis range

### PR DESCRIPTION
**Pull request by EmmadUsmani**
Closes https://github.com/metabase/metabase/issues/39596

### Description

The root cause was that we were including datums with null dimension values on timeseries, numeric (linear, log, power) and histogram charts, when they actually should be excluded from the dataset since we don't render them.

We fix this by filtering out those datums before applying other dataset transformations.

### Demo

Screenshot 2024-03-14 at 4 34 50 PM

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

**Comment by EmmadUsmani** [comment]
* **#40154** <a href="https://app.graphite.dev/github/pr/metabase/metabase/40154?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `echarts`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @EmmadUsmani and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: frontend/src/metabase/visualizations/echarts/cartesian/model/dataset.ts [hunk 1]
```diff
@@ -314,6 +315,23 @@ export const applySquareRootScaling = (value: RowValue): RowValue => {
   return value;
 };
 
+function filterNullDimensionValues(dataset: ChartDataset) {
+  // TODO show warning message
```
Comment (EmmadUsmani) [inline frontend/src/metabase/visualizations/echarts/cartesian/model/dataset.ts:319]: This is part of another issue

File: frontend/src/metabase/visualizations/echarts/cartesian/option/series.ts [hunk 2]
```diff
@@ -519,7 +519,7 @@ export const buildEChartsSeries = (
           return buildEChartsLineAreaSeries(
             seriesModel,
             seriesSettings,
-            chartModel.dataset,
+            chartModel.transformedDataset,
```
Comment (EmmadUsmani) [inline frontend/src/metabase/visualizations/echarts/cartesian/option/series.ts:522]: Not related to the main issue, but I noticed we weren't using `transformedDataset` here so I changed it

File: frontend/src/metabase/visualizations/echarts/cartesian/model/dataset.ts (hunk 1 above)
Comment (alxnddr) [reply frontend/src/metabase/visualizations/echarts/cartesian/model/dataset.ts:319]: Can you please update `null-category-value-formatting.json` visualization_settings.graph.x_axis.scale => "ordinal"?

**Review by alxnddr — APPROVED** [review]
Looks great! Can you please update one spec I mentioned before merging?

File: frontend/src/metabase/visualizations/echarts/cartesian/model/dataset.ts (hunk 1 above)
Comment (EmmadUsmani) [reply frontend/src/metabase/visualizations/echarts/cartesian/model/dataset.ts:319]: Yup let me do that

**Comment by EmmadUsmani** [comment]
### Merge activity

* **Mar 15, 8:43 PM EDT**: @EmmadUsmani started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/metabase/metabase/40154).

**Pull request merged**